app/routes/connectors/apps/CreateConnectorsForm.tsx

```diff
@@ -0,0 +1,162 @@
+import React, { FunctionComponent, useEffect } from 'react';
+
+import { Add } from '@mui/icons-material';
+import { pickBy, isEmpty } from 'lodash';
+import { Controller, useForm } from 'react-hook-form';
+import { useTranslation } from 'react-i18next';
+
+import { Select } from '@numaryhq/storybook';
+
+import {
+  buildForm,
+  connectorsConfig,
+} from '~/routes/connectors/apps/formBuilder';
+import Modal from '~/src/components/Wrappers/Modal';
+import { SnackbarSetter } from '~/src/contexts/service';
+import { useService } from '~/src/hooks/useService';
+import { ConnectorConfigFormProps } from '~/src/types/connectorsConfig';
+import { ApiClient, API_PAYMENT } from '~/src/utils/api';
+
+export const submit = async (
+  values: Partial<ConnectorConfigFormProps>,
+  connectorKey: keyof ConnectorConfigFormProps,
+  api: ApiClient,
+  snackbar: SnackbarSetter,
+  t: any
+) => {
+  try {
+    await api.postResource<any>(
+      `${API_PAYMENT}/connectors/${connectorKey}`,
+      values
+    );
+    // TODO find a better way to refresh the page
+    // await navigate('./apps', { replace: true });
+  } catch {
+    snackbar(
+      t('pages.apps.form.errors.errorOrDuplicate', {
+        connectorName: connectorKey,
+      })
+    );
+  }
+};
+
+export const CreateConnectorsForm: FunctionComponent = () => {
+  const { t } = useTranslation();
+  const { api, snackbar } = useService();
+
+  const formattedConnectorConfig = Object.keys(connectorsConfig).map((key) => ({
+    id: key,
+    label: key,
+  }));
+
+  const defaultValuesFromConfig = Object.entries(connectorsConfig).reduce(
+    (acc, [key, value]) => ({
+      ...acc,
+      [key]: Object.entries(value).reduce(
+        (acc, [key]) => ({
+          ...acc,
+          [key]: '',
+        }),
+        {}
+      ),
+    }),
+    {}
+  );
+
+  const {
+    trigger,
+    getValues,
+    formState: { errors, isValid },
+    control,
+    clearErrors,
+    reset,
+    watch,
+    handleSubmit,
+  } = useForm<any>({
+    mode: 'onChange',
+    defaultValues: {
+      connectorSelect: '',
+      ...defaultValuesFromConfig,
+    },
+  });
+
+  const connectorKey = watch('connectorSelect');
+
+  const onSave = async () => {
+    await trigger();
+
+    if (!isEmpty(errors)) {
+      return null;
+    }
+
+    // TODO change the type to it's real type
+    const sanitizedValues: Partial<ConnectorConfigFormProps> = pickBy(
+      getValues(connectorKey),
+      (value) => value.length > 0
+    );
+
+    await submit(sanitizedValues, connectorKey, api, snackbar, t);
```

Comment:
what happen in case of error ? do we have a snackbar somewhere ?

Reply:
Yes inside the submit